java-squid/src/main/java/org/sonar/java/parser/sslr/SyntaxTreeCreator.java

```diff
@@ -185,18 +149,28 @@ private InternalSyntaxToken visitTerminal(ParseNode node) {
       updateTokenPositionAndValue(node);
       TokenExpression ruleMatcher = (TokenExpression) node.getMatcher();
       tokenBuilder.setType(ruleMatcher.getTokenType());
+      isEof = ruleMatcher.getTokenType() == GenericTokenType.EOF;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)